api/handlers.go

```diff
@@ -289,13 +303,42 @@ func (a *API) Unmount(w http.ResponseWriter, r *http.Request) {
 
 	volName := volConfig.String()
 
+	ut := &config.UseMount{
+		Volume:   volName,
+		Reason:   lock.ReasonMount,
+		Hostname: a.Hostname,
+	}
+
+	if !volConfig.Unlocked {
+		a.RemoveStopChan(volName)
+
+		// XXX to doubly ensure we do not UNMOUNT something that is held elsewhere
+		// (presumably because it is mounted THERE instead), we refuse to unmount
+		// anything that doesn't acquire a lock. We also remove the TTL refresh
+		// before taking it so it is not cleared in the unlikely event the mount
+		// takes longer than the TTL. Re-establish the TTL on error only if it is in
+		// locked mode.
+		if err := a.Client.PublishUse(ut); err != nil {
```

Comment:
okay. Now, I understand that the whole purpose of this loop is to ensure that lock is still there. 

If so, Isn't enough just to check if publishUse fails on "ErrExists"